engine/src/main/java/org/terasology/world/block/family/AttachedToSurfaceFamily.java

```diff
@@ -15,47 +15,85 @@
  */
 package org.terasology.world.block.family;
 
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.Maps;
+import org.terasology.math.Rotation;
 import org.terasology.math.Side;
 import org.terasology.math.geom.Vector3i;
 import org.terasology.naming.Name;
+import org.terasology.registry.In;
 import org.terasology.world.BlockEntityRegistry;
 import org.terasology.world.WorldProvider;
 import org.terasology.world.block.Block;
+import org.terasology.world.block.BlockBuilderHelper;
 import org.terasology.world.block.BlockUri;
+import org.terasology.world.block.loader.BlockFamilyDefinition;
+import org.terasology.world.block.shapes.BlockShape;
 
 import java.util.Locale;
 import java.util.Map;
+import java.util.Set;
 
 /**
  */
+@RegisterBlockFamily("attachedToSurface")
+@BlockSections({"front", "left", "right", "back", "top", "bottom"})
+@MultiSections({
+        @MultiSection(name = "all", coversSection = "front", appliesToSections = {"front", "left", "right", "back", "top", "bottom"}),
+        @MultiSection(name = "topBottom", coversSection = "top", appliesToSections = {"top", "bottom"}),
+        @MultiSection(name = "sides", coversSection = "front", appliesToSections = {"front", "left", "right", "back"})})
 public class AttachedToSurfaceFamily extends AbstractBlockFamily {
+
+
     private Map<Side, Block> blocks = Maps.newEnumMap(Side.class);
     private Block archetype;
 
-    /**
-     * @param uri    The uri for the block group.
-     * @param blocks The set of blocks that make up the group. Front, Back, Left and Right must be provided - the rest is ignored.
-     */
-    public AttachedToSurfaceFamily(BlockUri uri, Map<Side, Block> blocks, Iterable<String> categories) {
-        super(uri, categories);
+    public AttachedToSurfaceFamily(BlockFamilyDefinition definition, BlockShape shape, BlockBuilderHelper blockBuilder) {
+        super(definition, shape, blockBuilder);
+        throw new UnsupportedOperationException("Freeform blocks not supported");
```

Comment:
Could you add a comment or reword this exception message to explain how a module developer could fix it? I'm not sure what would make this occur and would be very confused to receive this as written.

Reply:
Reword the exception. Be a bit more verbose.